### PR TITLE
Upgrade Yarn, Docusaurus, .etc

### PR DESCRIPTION
**Pull request by srmagura**
Note: Docusaurus doesn't yet support MDX v2 or React 18.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
+indent_size = 4
```
Comment (markerikson) [inline .editorconfig:6]: :speech_balloon: I'm more of a 2-space person these days, but less of a concern with Markdown I guess

**Comment by markerikson** [comment]
Hmm. Build failures are due to a complaint about a `_redirects` file. This is likely left over from me copy-pasting this from another Redux docs site, but not sure why it's complaining now. Can you double-check the config for this?

```
8:58:34 PM:   Error message
8:58:34 PM:   Command failed with exit code 1: yarn build && cp _redirects ./build (https://ntl.fyi/exit-code-1)
8:58:34 PM: ​
8:58:34 PM:   Error location
8:58:34 PM:   In build.command from netlify.toml:
8:58:34 PM:   yarn build && cp _redirects ./build
8:58:34 PM: ​
8:58:34 PM:   Resolved config
8:58:34 PM:   build:
8:58:34 PM:     base: /opt/build/repo/website
8:58:34 PM:     command: yarn build && cp _redirects ./build
8:58:34 PM:     commandOrigin: config
8:58:34 PM:     environment:
8:58:34 PM:       - REVIEW_ID
8:58:34 PM:     publish: /opt/build/repo/website/build
8:58:34 PM:     publishOrigin: config
```

**Comment by markerikson** [comment]
I think the issue is the `&& cp _redirects` part. We have that in some of the other Redux repos to deal with legacy URL paths, but the file doesn't exist here (and wouldn't be needed yet anyway).

**Comment by srmagura** [comment]
Should be good now. Had to make Netlify use a recent Node version and then fix some broken links.

**Comment by markerikson** [comment]
Sweet!

**Pull request merged**